src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final double side;
+
+    public Square(double side) {
+        this.side = side;
+        setColor(new ColorSupplier().getRandomColor());
```

Comment:
Avoid using a new instance of `ColorSupplier` directly in the constructor. You can pass the color value to the constructor of a figure instead of generating it inside figure classes.